### ISSUE TITLE
Are core.match :or clauses supported?

### ISSUE THREAD

**Issue by rbrw** [clj-kondo/clj-kondo#1657]
Version: 2022.04.08
(Invoked via lein alias expanding to `run -m clj-kondo.main`)

I noticed that an expression like this:
```clojure
(defn match-or
  [x]
  (cm/match
   [x]
   [(:or ["x" clause] ["y" clause])]
   clause
   :else :something))
```
provokes an unresolved symbol error:
```sh
$ lein kondo --lint src
src/kondo_match/core.clj:10:15: error: Unresolved symbol: clause
linting took 67ms, errors: 1, warnings: 0
Error encountered performing task 'run' with profile(s): 'base,system,user,provided,dev,kondo'
Suppressed exit
```
and I wondered whether that was currently expected, and if so, what was the preferred accommodation (the kondo profile only adds the kondo dependency).

Thanks

**Comment by borkdude** [comment]
It should work but perhaps you are accidentally using an older version of clj-kondo. The output that I'm seeing is:

<img width="638" alt="Screen Shot 2022-04-15 at 22 17 00" src="https://user-images.githubusercontent.com/284934/163627994-ce575962-13b8-45a6-9c37-2d7bb9e62fca.png">

which is also not correct, since the binding `clause` is used.

**Comment by borkdude** [comment]
Please provide the complete code including namespace form, clj-kondo config, etc. in a repo, then I can try to accurately repro, as I can't repro it using only this snippet.

**Comment by rbrw** [comment]
Hmm, so I went back and re-read the documentation, and noticed the recommended `--dependencies` argument.  I'm not sure I'd seen that last time I read them, but adding it appears to fix the problem.  If that seems like the appropriate fix, please feel free to close this.

**Comment by rbrw** [comment]
Hah, OK, so I think I just misinterpreted what the docs were saying, and that option only "fixes" the error because it doesn't scan the actual code anymore.

**Comment by borkdude** [comment]
I think this is a bug in clj-kondo. All I'm asking is a repro using an ns form + code + file extension (to indicate if it's clojure or clojurescript). The support for core.match is built into clj-kondo, it's not coming from an external config.

**Comment by rbrw** [comment]
To reproduce the problem, just two files were changed in a `lein new app kondo-match` tree (lein 2.9.8).
`project.clj`:
```
(defproject kondo-match "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "EPL-2.0 OR GPL-2.0-or-later WITH Classpath-exception-2.0"
            :url "https://www.eclipse.org/legal/epl-2.0/"}
  :dependencies [[org.clojure/clojure "1.11.0"]]
  :main ^:skip-aot kondo-match.core
  :target-path "target/%s"
  :profiles {
             :kondo {:dependencies [[clj-kondo "2022.04.08"]]}
             :uberjar {:aot :all
                       :jvm-opts ["-Dclojure.compiler.direct-linking=true"]}}
  :aliases
  {"kondo" ["with-profiles" "+kondo" "run" "-m" "clj-kondo.main"]})
```
and `src/kondo_match/core.clj`:
```
(ns kondo-match.core
  (:gen-class)
  (:require
   [core.match :as cm]))

(defn match-or
  [x]
  (cm/match
   [x]
   [(:or ["x" clause] ["y" clause])]
   clause
   :else :something))

(defn -main
  "I don't do a whole lot ... yet."
  [& _args]
  (println "Hello, World!"))
```
After those changes I see the error:
```
$ lein kondo --lint src
src/kondo_match/core.clj:10:15: error: Unresolved symbol: clause
linting took 71ms, errors: 1, warnings: 0
```

**Comment by borkdude** [comment]
Thanks.
